### DISCUSSION TITLE
Decouple Admin Dashboard from Backend

### DISCUSSION THREAD

**Discussion by revlov** [medusajs/medusa#5973] [Feature Requests]
Refering to the official [Admin Dashboard Quickstart](https://docs.medusajs.com/admin/quickstart#overview), it says:

> The admin dashboard is installed on the Medusa backend. The admin dashboard starts when you start the Medusa backend. This also means you can later deploy the Medusa backend along with the admin dashboard on the same hosting.

I wonder if it is actually possible to decouple the **Admin Dashboard** (as is) from the backend, having a folder for every servable client, structured like this:

```
apps/
  admin/
    src/
    package.json
    ...
  backend/
    src/
    package.json
    ...
  storefront/
    src/
    package.json
    ...
```

I truly understand that Medusa is a headless framework, and I can make my own admin panel, as I wish. However, I want to go for the default templates, which already are very robust.

As I understand, the admin dashboard is a plugin that has to be added to `medusa-config.js`, which makes it depend on the backend project and its build process?! Or can I run the admin dashboard independently as a client project?

If anybody can shed some light on this, I would really appreciate.

**Comment by chrislaai** [comment]
It seems that a true headless e-commerce framework will have a separate admin repo. Since Medusa has given up the separate Admin repo, I feel that the current Medusa is not completely headless. I am not saying that it cannot be headless, but it is the default the API and admin are bound together and in some cases require each other. I also kinda hope Admin and API could be totally split, but it doesn't seem like that's the way things are heading.

**Comment by liviktech** [comment]
After reading the docs and analyzing the flow, I can see that when taking applications to production, the admin and backend builds will be separated into two different folders in the `.medusa` folder. So technically, we can deploy them separately.

#### Current State
Indeed, at the time of development both the admin panel and backend are tightly coupled.

I'm able to add pages and routes with the SDK and Widgets. As per my understanding, these will be included in the production build with the help of `admin-bundler` when we take the build.

However, customization on the built-in sections - I haven't found any ways to do this so far, like we do with the SDK and Widgets. (A workaround would be to create new pages for those features we could think of.)

#### Implementation Approach

@revlov, I'm able to get the folder structure you've mentioned. I don't know what issues it can create in the future, like getting the latest updates to the admin panel, or adding new features...

Here are the steps I followed:

1. **Admin Panel**: I copied the dashboard folder from the main medusa repo - `medusa/packages/admin/dashboard`
2. **Backend**: I cloned the [medusa-starter-default](https://github.com/medusajs/medusa-starter-default) for the backend which will be used with the CMA command
3. **Storefront**: Using the [nextjs-starter-medusa](https://github.com/medusajs/nextjs-starter-medusa) for the storefront

#### Current Folder Structure
<img width="266" height="255" alt="image" src="https://github.com/user-attachments/assets/720aeb6d-94c1-4683-b59d-0aee3f20d4e5" />

## Disclaimer

I'm still learning about the quirks and turns in the repository. So, I don't recommend using this approach in production, but I think it gives you the core idea. I hope it's useful to you guys as well.

---

**Note**: This approach allows for separate development of admin and backend, but comes with potential maintenance challenges and may not receive automatic updates from the main Medusa repository.

**Comment by lindalinya** [comment]
1. Disable the native admin dashboard in the config according to [Admin-configuration-disabled](https://docs.medusajs.com/learn/configurations/medusa-config#disable)
2. Do anything what you want to do.
